.github/workflows/integration-test.yml

```diff
@@ -12,21 +12,49 @@ env:
 jobs:
   tests:
     name: Run integration tests
-    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest , windows-latest ]
+        jdk: [ 11, 17 ]
+    runs-on: ${{ matrix.os }}
+
     steps:
       - name: Checkout Branch
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
+
+      - name: Set up JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.jdk }}
 
       - uses: browser-actions/setup-geckodriver@latest
       - run: geckodriver --version
 
-      - uses: browser-actions/setup-firefox@latest
-      - run: firefox --version
-          
-      - name: Download OpenSearch Security Plugin
+      - name: Set up Firefox browser for Linux
+        if: ${{ runner.os == 'Linux' }}
+        uses: browser-actions/setup-firefox@latest
```

Comment:
Replace `@latest` with either a specific version tag or commit id.

> We strongly recommend that you include the version of the action you are using by specifying a Git ref, SHA, or Docker tag. If you don't specify a version, it could break your workflows or cause unexpected behavior when the action owner publishes an update.

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsuses

Reply:
Note; this applies to a couple of different places in these changes

Reply:
Thanks for the review Peter! I have changed the firfox one to a specific version. But for the gecko I think it only has the version control of `latest`.